website_snippet_product_category/static/src/js/frontend.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Tecnativa - Alexandre Díaz
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+odoo.define("website_snippet_product_category.s_product_category", function (require) {
+    "use strict";
+
+    var core = require("web.core");
+    var sAnimation = require("website.content.snippets.animation");
+
+    var _t = core._t;
+
+    sAnimation.registry.js_product_category = sAnimation.Class.extend({
```

Comment:
If I'm not wrong, the correct class to extend is `web_editor.snippets.options.SnippetOption`, which includes a `cleanForSave()` method that should empty this widget before hitting "Save". It would solve this problem:

1. User edits website and drops this snippet.
1. It downloads current categories.
1. User saves.
1. Those categories are saved as part of the page.
1. eCommerce categories change, but this page has a hardcoded collection in this snippet.
1. A visitor browses this page. He sees briefly the old categories before the rpc call is completed.

This would possibly imply a refactor that maybe is not worth the effort, so let's leave this as it is. I'd label this module as Alpha category though.

Reply:
web_editor.snippets.options.SnippetOption is used for snippet options
sAnimation is used for snippet initialization (and other website elements)

This snippet doesn't allow to modify their content. The only way to change it is on the 'backend' or using the snippet options.